wicket/src/state/update.rs

```diff
@@ -103,6 +127,162 @@ impl RackUpdateState {
+                        );
+                    }
+                    UpdateNormalEventKind::InstallinatorEvent(
+                        _completion_event_kind,
+                    ) => {
+
+                        // TODO: Do we want to update state
+                        // with intermediate events?
+                        //
+                        // This would be useful for reporting progress
+                        // percentages, but that's about it.
+                    }
+                },
+                UpdateEventKind::Terminal(terminal) => match terminal {
+                    UpdateTerminalEventKind::SpResetFailed { .. } => {
+                        let known = Some(id.sp_known_artifact_kind());
+                        update_artifact_state(
+                            items,
+                            known,
+                            UpdateState::Failed,
+                        );
+                    }
+                    UpdateTerminalEventKind::ArtifactUpdateFailed {
+                        artifact,
+                        ..
+                    } => {
+                        let known = artifact_to_known_artifact_kind(artifact);
+                        update_artifact_state(
+                            items,
+                            known,
+                            UpdateState::Failed,
+                        );
+                    }
+                },
+            }
+        }
+
+        // Mark any known artifacts as updating
+        let Some(state) = &log.current else {
+            return;
+        };
+        use UpdateStateKind::*;
+        let known = match &state.kind {
+            WaitingForProgress { .. } => {
+                // Nothing to do here
+                None
+            }
+            ResettingSp => Some(id.sp_known_artifact_kind()),
+            SendingArtifactToMgs { artifact } => {
+                artifact_to_known_artifact_kind(&artifact)
+            }
+            PreparingForArtifact { artifact, .. } => {
+                artifact_to_known_artifact_kind(&artifact)
+            }
+            ArtifactDownloadProgress { kind, .. } => {
+                artifact_kind_to_known_kind(kind)
+            }
+            ArtifactWriteProgress { kind, .. } => {
+                artifact_kind_to_known_kind(kind)
+            }
+            WaitingForStatus { artifact } => {
+                artifact_to_known_artifact_kind(&artifact)
+            }
+            WaitingForTrampolineImageDelivery { artifact, .. } => {
+                artifact_to_known_artifact_kind(&artifact)
+            }
+            SettingHostPowerState { .. }
+            | InstallinatorFormatProgress { .. }
+            | SettingInstallinatorOptions
+            | SettingHostStartupOptions => {
+                // Should we bother doing something here?
+                None
+            }
+        };
+        update_artifact_state(items, known, UpdateState::Updating);
```

Comment:
Per discussion, filed https://github.com/oxidecomputer/omicron/issues/2578 to show a progress percentage here.

Reply:
Thank you!